narwhals/utils.py

```diff
@@ -1957,6 +1945,41 @@ def decorate(init_child: _Constructor[_T, P, R2], /) -> _Constructor[_T, P, R2]:
     return decorate
 
 
+def ensure_type(obj: Any, /, *valid_types: type[Any], param_name: str = "") -> None:
+    """Validate that an object is an instance of one or more specified types.
+
+    Parameters:
+        obj: The object to validate.
+        *valid_types: One or more valid types that `obj` is expected to match.
+        param_name: The name of the parameter being validated.
+            Used to improve error message clarity.
+
+    Raises:
+        TypeError: If `obj` is not an instance of any of the provided `valid_types`.
+
+    Examples:
+        >>> from narwhals.utils import ensure_type
+        >>> ensure_type(42, int, float)
+        >>> ensure_type("hello", str)
+        >>> ensure_type("hello", int, param_name="test")
+        Traceback (most recent call last):
+            ...
+        TypeError: Expected 'int', got: 'str'
+            test='hello'
+                 ^^^^^^^
+    """
+    if not isinstance(obj, valid_types):  # pragma: no cover
+        tp_names = " | ".join(tp.__name__ for tp in valid_types)
+        msg = f"Expected {tp_names!r}, got: {type(obj).__name__!r}"
+        if param_name:
+            left_pad = " " * 4
+            val = repr(obj)
```

Comment:
My bad on this part, but probably a good idea to set some limit on the repr

```suggestion
            val = repr(obj)
            truncate_limit = 20
            if len(val) > truncate_limit:
                val = f"{type(obj).__name__}(...)"
```

We don't want this 😄 

```py
import polars as pl

data = {"a": [1, 2, 3], "b": [4, 5, 6]}
df = pl.DataFrame(data)

val = repr(df)
underline = "^" * len(val)
>>> print(f"{val}\n{underline}")
shape: (3, 2)
┌─────┬─────┐
│ a   ┆ b   │
│ --- ┆ --- │
│ i64 ┆ i64 │
╞═════╪═════╡
│ 1   ┆ 4   │
│ 2   ┆ 5   │
│ 3   ┆ 6   │
└─────┴─────┘
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```